### ISSUE TITLE
Mapping dash-leaflet controls to other buttons outside the map

### ISSUE THREAD

**Issue by prl900** [thedirtyfew/dash-leaflet#102]
Just drawing attention here to [this question](https://stackoverflow.com/questions/68510848/mapping-dash-leaflet-controls-to-other-buttons-outside-the-map) I just posted on stackoverflow.

I have noticed I have been the first person to use the `dash-leaflet` tag in stackoverflow, so I'm not sure if this community is active there.

Thanks

**Comment by emilhe** [comment]
Thanks for the question! I believe that for general questions (i.e. not bug reports/feature requests, they should go here), stackoverflow is indeed the correct forum. I wasn't aware that a `dash-leaflet` tag was available :)

**Comment by emilhe** [comment]
This feature has been implemented as part of `dash-leaflet==0.1.18`. Hence you can now invoke the drawing tools via Dash callbacks.  Here is a small example,

```
import dash_html_components as html
import dash_leaflet as dl
from dash.dependencies import Output, Input
from dash_extensions.enrich import DashProxy, MultiplexerTransform

# Mode definitions.
draw_modes = ["marker", "polygon", "polyline", "rectangle", "circle", "circlemarker"]
edit_modes = ["edit", "remove"]
# Create example app.
app = DashProxy(transforms=[MultiplexerTransform()], prevent_initial_callbacks=True)
app.layout = html.Div(
    [
        dl.Map(center=[56, 10], zoom=4, children=[
            dl.TileLayer(), dl.FeatureGroup([
                dl.EditControl(id="edit_control"),
                dl.Marker(position=[56, 10])]),
        ], style={'width': '100%', 'height': '50vh', 'margin': "auto", "display": "inline-block"},
               id="map")
    ] +
    # Mode buttons.
    [html.Div("Draw")] +
    [html.Button(mode, id=mode) for mode in draw_modes] +
    [html.Div("Edit")] +
    [html.Button(mode, id=mode) for mode in edit_modes] +
    # Action buttons.
    [
        html.Div("Actions"),
        html.Button("polygon -> delete last point", id="poly_delete"),
        html.Button("polygon -> finish", id="poly_finish"),
        html.Button("remove -> clear all", id="clear_all")
    ])
# Create draw mode callbacks.
for mode in draw_modes:
    app.callback(Output("edit_control", "drawToolbar"), Input(mode, "n_clicks"))(
        lambda n_clicks, m=mode: dict(mode=m, n_clicks=n_clicks))
# Create edit mode callbacks.
for mode in edit_modes:
    app.callback(Output("edit_control", "editToolbar"), Input(mode, "n_clicks"))(
        lambda n_clicks, m=mode: dict(mode=m, n_clicks=n_clicks))
# Create a few action callbacks, but not all - there are too many combinations...
app.callback(Output("edit_control", "drawToolbar"), Input("poly_delete", "n_clicks"))(
    lambda n_clicks: dict(mode="polygon", n_clicks=n_clicks, action="delete last point"))
app.callback(Output("edit_control", "drawToolbar"), Input("poly_finish", "n_clicks"))(
    lambda n_clicks: dict(mode="polygon", n_clicks=n_clicks, action="finish"))
app.callback(Output("edit_control", "editToolbar"), Input("clear_all", "n_clicks"))(
    lambda n_clicks: dict(mode="remove", n_clicks=n_clicks, action="clear all"))

if __name__ == '__main__':
    app.run_server(port=7781, debug=True)
```
